apps/desktop/src/app/sync/sources/s3.ts

```diff
@@ -0,0 +1,140 @@
+  resumable = false;
+  client?: S3Client;
+  bucket?: string;
+
+  constructor(data?: ConnectorSettings) {
+    // eslint-disable-next-line no-empty-function
+  }
+
+  getParameters(): Observable<Field[]> {
+    return of([
+      {
+        id: 'region',
+        label: 'Region',
+        type: 'text',
+        help: 'e.g. us-east-1',
+        required: true,
+      },
+      {
+        id: 'bucket',
+        label: 'Bucket',
+        type: 'text',
+        required: true,
+      },
+      {
+        id: 'access_key_id',
+        label: 'Access key id',
+        type: 'text',
+        required: true,
+      },
+      {
+        id: 'secret_access_key',
+        label: 'Secret access key',
+        type: 'text',
+        required: true,
+      },
+    ]);
+  }
+
+  handleParameters(params: ConnectorParameters) {
+    this.bucket = params.bucket;
+    this.client = new S3Client({
+      region: params.region,
+      credentials: {
+        accessKeyId: params.access_key_id,
+        secretAccessKey: params.secret_access_key,
+      },
+    });
+  }
+
+  goToOAuth() {
+    // eslint-disable-next-line no-empty-function
+  }
+
+  authenticate(): Observable<boolean> {
+    return this.client && this.bucket ? of(true) : of(false);
+  }
+
+  getFiles(query?: string, pageSize?: number): Observable<SearchResults> {
+    if (query) {
+      // S3 API doesn't have any command to filter by keywords.
+      // As a workaround we retrieve all the objects and do the filtering ourselves.
+      pageSize = MAX_PAGE_SIZE;
+    }
+    return this._getFiles(query, pageSize);
+  }
+
+  private _getFiles(query?: string, pageSize: number = 50, nextPage?: string | number): Observable<SearchResults> {
+    const command = new ListObjectsV2Command({
+      Bucket: this.bucket || '',
+      MaxKeys: pageSize,
+      ContinuationToken: nextPage as string,
+    });
+    return from(this.client!.send(command)).pipe(
+      map((results) => ({
+        items: (results.Contents || [])
+          .filter((item) => this.filterResult(item, query))
+          .map((result) => this.mapResult(result)),
+        nextPage:
+          results.IsTruncated && results.NextContinuationToken
+            ? this._getFiles(query, pageSize, results.NextContinuationToken)
+            : undefined,
+      })),
+    );
+  }
+
+  private mapResult(result: _Object) {
+    const delimiter = '/'; // Only default delimiter is supported at the moment
+    const title = (result.Key || '').split(delimiter).pop();
+    return {
+      originalId: result.Key || '',
+      title: title || '',
+      uuid: '',
+      metadata: {},
+      status: FileStatus.PENDING,
+    };
+  }
+
+  private filterResult(result: _Object, query?: string) {
+    if (query && result.Key) {
+      const regex = new RegExp(`(${query})`, 'i');
+      return regex.test(result.Key);
+    }
+    return true;
+  }
+
+  download(resource: SyncItem): Observable<Blob> {
+    const command = new GetObjectCommand({
```

Comment:
Actually, the objective with Amazon S3 (and with Google Cloud Service) is not to download the physical and then upload it back to Nuclia like we did with DropBox or GDrive.
We want to use them as cloud file storage, so we want to use the Nuclia cloud file support, meaning we will create a FileField by providing its direct URL to Amazon S3 (and also some extra_headers to provide an auth token or whatever Amazon S3 in order to give access to the file)